src/Common/src/System/IO/StreamAsyncHelper.cs

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO
+{
+    /// <summary>Provides support for implementing asynchronous operations on Streams.</summary>
+    internal sealed class StreamAsyncHelper
+    {
+        private SemaphoreSlim _asyncActiveSemaphore;
+        private Task _activeReadWriteTask;
+        private readonly Stream _stream;
+
+        internal StreamAsyncHelper(Stream stream)
+        {
+            Debug.Assert(stream != null);
+            _stream = stream;
+        }
+
+        private SemaphoreSlim EnsureAsyncActiveSemaphoreInitialized()
+        {
+            // Lazily-initialize _asyncActiveSemaphore.  As we're never accessing the SemaphoreSlim's
+            // WaitHandle, we don't need to worry about Disposing it.
+            return LazyInitializer.EnsureInitialized(ref _asyncActiveSemaphore, () => new SemaphoreSlim(1, 1));
+        }
+
+        internal IAsyncResult BeginRead(byte[] buffer, int offset, int count, AsyncCallback callback, object state)
+        {
+            if (!_stream.CanRead) throw __Error.GetReadNotSupported();
```

Comment:
Good catch: this file got deleted during work I did for this PR. This is just bringing it back to the same location.

It appears that the file never got formatted for some reason. I'll run the tools again.